shielded-pool/src/component.rs

```diff
@@ -88,12 +88,23 @@ impl Component for ShieldedPool {
             .unwrap();
         }
 
-        self.compact_block.height = 0;
+        self.compact_block.height = self
+            .state
+            .get_block_height()
+            .await
+            .expect("block height must be set");
+
```

Comment:
In `init` we should be hardcoding the block height as 0; I'm not sure that the block height will be set at this point, because there hasn't been a `BeginBlock`.  The 0th CompactBlock is special-cased out of the genesis app state in `InitChain`, so it doesn't actually correspond to a real Tendermint block, it's just a way to hook the genesis state up with the client sync code.

Reply:
Ah, I see, that makes sense.